### PR TITLE
Underworld Maze Improvements

### PR DESCRIPTION
**Pull request by V2LenKagamine**
## About The Pull Request
Technical Changes:
 - changes `underworldcoinspawns` to `underworldspiritspawns`
 - changes unused `underworldcoins` list to `underworld_coinpull_locs`
 - client verb `decend` now gives an error if it cant find a location to spawn the spirit, and may pick a random one.
 - New Coin-spawner, ~~spawns coins for the maze~~, and pulls souls to the carrage on finding them!
 - - Has been mapped into the maze.
 - - New marker to set where it can pull you, also mapped in
 - The Maze is now slightly bigger.
 - The Soul Lantern is now *slightly* larger and blue to be easier to see and match the lights.
 - Under-arena now takes into account when you died.
## Why It's Good For The Game

I have heard from many the maze sucks, so I made it so that:
 - ~~Coins in the maze respawn periodically~~
 - An object to spawn them periodically has been created
 - Coins from said object move you to the carrageman, or nearby them
 - The lantern doesnt suck.

## Pre-Merge Checklist

- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

### CODE REVIEW

File: code/modules/underworld/underworld.dm [hunk 1]
```diff
@@ -169,20 +160,61 @@
...
+		to_chat(fool,span_alertwarning("\The [src] swiftly drags you under; but leaves you with \the [toll]!"))
+	else
+		to_chat(fool,span_alertwarning("\The [src] swiftly drags you under; but leaves \the [toll] at your feet!"))
+	set_coin_taken()
+
+/obj/structure/underworld/coinspawner/proc/set_coin_taken()
+	has_coin = FALSE
+	icon_state = "the_hand"
+	desc = "A hand?"
+	addtimer(CALLBACK(src,TYPE_PROC_REF(/obj/structure/underworld/coinspawner,regenerate_coin)),5 MINUTES)
```
Comment (dwasint) [inline code/modules/underworld/underworld.dm:205]: way to short, the reason they don't respawn is we don't want players flooding the game later in the round when things are dilluted. I don't want them respawning.

Comment (V2LenKagamine) [reply code/modules/underworld/underworld.dm:205]: Will bump to 20 minutes

Comment (dwasint) [reply code/modules/underworld/underworld.dm]: The concept of them respawning is the issue, it means the flow of people will almost always be the same which we do not want as it trivializes death, as said remove this

Comment (V2LenKagamine) [reply code/modules/underworld/underworld.dm]: The main problem with them *not* respawning is that people just get stuck in underworld waiting for 9 years until 10 min coin pity, this allows the adventurous ones to venture into the maze and *hopefully* find a coin; should arena be busy or the *pit* broken, there are only a few coin spawners

Comment (V2LenKagamine) [reply code/modules/underworld/underworld.dm]: Removed.

File: code/_onclick/hud/ghost.dm [hunk 2]
```diff
@@ -72,13 +74,16 @@
...
 		if(alert("Travel with the boatman?", "", "Yes", "No") == "Yes")
 			verbs -= /client/proc/descend
-			for(var/turf/spawn_loc in GLOB.underworldspiritspawns)
-				var/mob/living/carbon/spirit/O = new /mob/living/carbon/spirit(spawn_loc)
-				O.livingname = G.name
-				O.ckey = G.ckey
-				ADD_TRAIT(O, TRAIT_PACIFISM, TRAIT_GENERIC)
-				SSdeath_arena.add_fighter(O)
-				SSdroning.area_entered(get_area(O), O.client)
+			if(!GLOB.underworldspiritspawns.len) //That cant be good.
```
Comment (dwasint) [inline code/_onclick/hud/ghost.dm:77]: use length() instead of .len

Comment (V2LenKagamine) [reply code/_onclick/hud/ghost.dm]: Donezers

**Pull request merged**